plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -63,6 +63,7 @@ newtype ExMemory = ExMemory Integer
 -- Yes, I also hate that.
 deriving newtype instance PrettyDefaultBy config Integer => PrettyBy config ExMemory
 
+-- TODO: 'Integer's are not particularly fast. Should we use @Int64@?
```

Comment:
A quick experiment suggests that this would be about 15-20% faster: see https://jira.iohk.io/browse/SCP-1973.  There may be a danger of DOS attacks though, so we should think about this carefully.

Reply:
Thanks for checking and creating an issue!